### PR TITLE
manifest: update sof module

### PR DESCRIPTION
**Pull request by iuliana-prodan**
Update SOF module to latest main.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

### CODE REVIEW

**Comment by iuliana-prodan** [comment]
> @iuliana-prodan so I've just had notifications from Zephyr CI that the SOF module main branch update failed. I'm not sure if this is due to the change to `main` from `master` or something else. It cant be a build failure since the west manifest is not updated and still pointing to the old branch and commit.

After adding this commit, that updates the manifest with the commit id from main branch, I did a `west update sof` and it brought me the latest SOF module.

**Comment by marc-hb** [comment]
> so I've just had notifications from Zephyr CI that the SOF module main branch update failed. 

Can you share a link?  I don't see any failure here.

**Comment by lgirdwood** [comment]
> > so I've just had notifications from Zephyr CI that the SOF module main branch update failed.
> 
> Can you share a link? I don't see any failure here.

Thank you @marc-hb !

https://github.com/zephyrproject-rtos/sof/actions/runs/1071625778

https://github.com/zephyrproject-rtos/sof/actions/runs/1071625775

https://github.com/zephyrproject-rtos/sof/actions/runs/1071625776

**Comment by marc-hb** [comment]
Thanks @lgirdwood for the links. What's happening is the `zephyrproject-rtos/sof` fork[*] tries to run the same github actions than the original `thesofproject/sof`, because it shares the same `.github/` directory. Then it fails because of configuration differences. 

["fork" is how github calls any remote clone whether it actually branches out or not. There is no formal definition of the word "fork". This is discussing a github problem so I'm using github terminology]

The fix is to simply disable github actions on zephyrproject-rtos/sof  https://docs.github.com/en/github/administering-a-repository/managing-repository-settings/disabling-or-limiting-github-actions-for-a-repository.

We have no interest in maintaining github actions that work across forks, especially not when that "fork" is not actually branching out and would re-run the same github actions on the same commits on which they have already been run in the original repo. No value in that.

**Comment by lgirdwood** [comment]
@marc-hb thanks, so you mean this is an AR for Zephyr team to disable locally for SOF module in the GH settings ?

**Comment by marc-hb** [comment]
> @marc-hb thanks, so you mean this is an AR for Zephyr team to disable locally for SOF module in the GH settings ?

Yes, exactly that - and probably for most other zephyrproject-rtos/forks using github actions too.

It looks like github enables this by default which makes no sense: https://github.community/t/stop-github-actions-running-on-a-fork/17965/3. Maybe they'll stop enabling this by default once some hackers figure out how to abuse it? https://www.infoq.com/news/2021/04/GitHub-actions-cryptomining/

**Review by dbaluta — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by iuliana-prodan** [comment]
Can somebody, please, merge this - is pending for a week now.

cc: @nashif @MaureenHelm @lgirdwood @dbaluta @dcpleung @dleach02

**Comment by lgirdwood** [comment]
@iuliana-prodan merging is blocked as the buildkite check is failing. Can you check and fix.

**Comment by dleach02** [comment]
there is a build failure for samples/subsys/audio/sof/sample.audio.sof that needs to be resolved.. not sure if this is associated with the manifest change or not

**Comment by iuliana-prodan** [comment]
@lgirdwood @dleach02 I've looked on the failure and it seem like the rimage submodule is not brought into modules/audio/sof:
```
-- Git submodules update
fatal: Needed a single revision
Unable to find current revision in submodule path '../rimage'
CMake Error at /workdir/modules/audio/sof/scripts/cmake/git-submodules.cmake:47 (message):
  git submodule update --init failed with 1, please checkout submodules
Call Stack (most recent call first):
  CMakeLists.txt:23 (include)
```
I still couldn't figure out, yet, why is not bringing the rimage submodule.

cc: @carlescufi @nashif @dcpleung

**Comment by nashif** [comment]
> @nashif do we need to do anything else when pushing commits from SOF main to SOF module main ? We are just do a synchronisation ?

Usually, you need to push changes as a PR, you also need to keep the local changes of the module. 
This does not seem to be the case here.

File: west.yml
Comment (dcpleung) [inline west.yml:198]: Try adding `submodules: true' and see if it works?

Comment (nashif) [reply west.yml:198]: do not think this is the problem, the way the module has been updated is wrong, I am looking into it right now.

**Comment by nashif** [comment]
@lgirdwood we need this in the SOF tree: https://github.com/thesofproject/sof/pull/4594

**Comment by lgirdwood** [comment]
> @lgirdwood we need this in the SOF tree: [thesofproject/sof#4594](https://github.com/thesofproject/sof/pull/4594), then we could just have a clean fork of SOF

Thanks @nashif SOF https://github.com/thesofproject/sof/pull/4594 now merged.

**Comment by lgirdwood** [comment]
@iuliana-prodan are you able to shed any light on the rimage issue ? @kv2019i fyi.
Could this be a result of changing branch ```master``` -> ```main``` for rimage after the initial rimage commit ID and branch was added to west ?

**Comment by iuliana-prodan** [comment]
> @iuliana-prodan are you able to shed any light on the rimage issue ? @kv2019i fyi.
> Could this be a result of changing branch `master` -> `main` for rimage after the initial rimage commit ID and branch was added to west ?

@lgirdwood the problem is this commit: https://github.com/thesofproject/sof/pull/3993/commits/c6b6d74aef3f652c66f13a4d145c14b1f661807b that changes the rimage repo URL to a relative path.

On Zephyr project we have a fork of SOF, but not of rimage repo.
That's why the rimage submodule is not brought with SOF.

A solution is to revert this commit just in the SOF fork: set the rimage URL as https://github.com/thesofproject/rimage.git.

Or clone/fork rimage repo on Zephyr project.

I prefer the 1st fix.
Do you agree with this?

cc: @marc-hb @nashif @kv2019i @dleach02 @MaureenHelm @dcpleung @carlescufi @dbaluta

**Comment by lgirdwood** [comment]
> > @iuliana-prodan are you able to shed any light on the rimage issue ? @kv2019i fyi.
> > Could this be a result of changing branch `master` -> `main` for rimage after the initial rimage commit ID and branch was added to west ?
> 
> @lgirdwood the problem is this commit: [thesofproject/sof@c6b6d74](https://github.com/thesofproject/sof/commit/c6b6d74aef3f652c66f13a4d145c14b1f661807b) that changes the rimage repo URL to a relative path.
> 
> On Zephyr project we have a fork of SOF, but not of rimage repo.
> That's why the rimage submodule is not brought with SOF.
> 
> A solution is to revert this commit just in the SOF fork: set the rimage URL as https://github.com/thesofproject/rimage.git.
> 
> Or clone/fork rimage repo on Zephyr project.
> 
> I prefer the 1st fix.
> Do you agree with this?
> 
> cc: @marc-hb @nashif @kv2019i @dleach02 @MaureenHelm @dcpleung @carlescufi @dbaluta

Ok, @marc-hb is back from vacation today - lets wait for his feedback as he is working on the Zephyr integration too so this may be part of his plans (and possible missing further updates). I think he was looking at better integration with west.

**Comment by marc-hb** [comment]
>  the problem is this commit: thesofproject/sof@c6b6d74 that changes the rimage repo URL to a relative path.

Interesting but why does it fail only now? `sof/.gitmodules` already has a relative path right now in Zephyr even before this PR: https://github.com/zephyrproject-rtos/sof/blob/779f28ed465c0/.gitmodules

This PR does NOT change `sof/.gitmodules`, it's been like this in Zephyr since May 24:
https://github.com/zephyrproject-rtos/zephyr/pull/35559/files

`Unable to find current revision` also does not really sound like a failure to find the entire submodule, does it? I will try to reproduce.


> A solution is to revert this commit just in the SOF fork: set the rimage URL as https://github.com/thesofproject/rimage.git

I believe upstream Zephyr has a policy not to point at external git repos (that can come and go).

From https://docs.zephyrproject.org/latest/guides/modules.html#module-repositories
> All modules included in the default manifest shall be hosted in repositories under the zephyrproject-rtos GitHub organization.

**Comment by iuliana-prodan** [comment]
> > the problem is this commit: [thesofproject/sof@c6b6d74](https://github.com/thesofproject/sof/commit/c6b6d74) that changes the rimage repo URL to a relative path.
> 
> Interesting but why does it fail only now? `sof/.gitmodules` already has a relative path right now in Zephyr even before this PR: https://github.com/zephyrproject-rtos/sof/blob/779f28ed465c0/.gitmodules
> 
> This PR does NOT change `sof/.gitmodules`, it's been like this in Zephyr since May 24:
> https://github.com/zephyrproject-rtos/zephyr/pull/35559/files
> 

I don't see any tests run on https://github.com/zephyrproject-rtos/sof/tree/779f28ed465c03899c8a7d4aaf399856f4e51158.
The last pull request (before the current falling one) is https://github.com/zephyrproject-rtos/sof/pull/11 and I don't see any tests or builds.
Are you sure this really worked before the relative path change?

**Comment by marc-hb** [comment]
Audio tests in shard 3/20 passed in https://buildkite.com/zephyr/zephyr/builds/31091#a0d6c1c0-1509-4fd6-b8b1-42389e2046fe for SOF update https://github.com/zephyrproject-rtos/zephyr/pull/35559/files

> I don't see any tests run on https://github.com/zephyrprojectrtos/sof/tree/779f28ed465c03899c8a7d4aaf399856f4e51158.

There should be no CI on the `zephyrprojectrtos/sof/` "fork", see July 27th discussion above. CI is on upstream `thesofproject/sof` and on `zephyrprojectrtos/zephyr/`

**Comment by marc-hb** [comment]
It could be caused by https://github.com/thesofproject/sof/commit/f6c71c21d03b8111 that added a dependency on the sof-logger, stay tuned.

**Comment by marc-hb** [comment]
> It could be caused by thesofproject/sof@f6c71c2 that added a dependency on the sof-logger and smex, stay tuned.

Confirmed.

The spurious sof-logger -> rimage dependency is fixed by https://github.com/thesofproject/sof/pull/4669

I tested this more and found that smex does not depend on rimage either. smex depends only on a small `ext_manifest.h` file which can be copied (and was copied into `sof/zephyr` by https://github.com/zephyrproject-rtos/sof/pull/7). I worked around this smex issue by copying `ext_manifest.h` manually into some made-up directory and that was enough to build `zephyr/samples/subsys/audio/sof` fully and successfully. Will hopefully find a clean way to fix this smex issue tomorrow.

tl;dr: the Zephyr build should still not need any rimage clone.

**Comment by dbaluta** [comment]
@marc-hb @iuliana-prodan SOF patches are now merged. Is there any work needed for this to get in?

**Comment by marc-hb** [comment]
> @marc-hb @iuliana-prodan SOF patches are now merged.

This fixed only one rimage dependency.

> Is there any work needed for this to get in?

I'm finalizing another and last SOF PR to remove the remaining rimage dependencies.

**Comment by dbaluta** [comment]
@marc-hb  I like to understand if we can hep with that, as we figured out today that zephyr merge window is Friday :)

**Comment by marc-hb** [comment]
They were actually not just 2 but 3 rimage-related failures in this PR:

- Fail1 sof-logger misses rimage. Already fixed by https://github.com/thesofproject/sof/pull/4669

- Fail2.1 smex misses   `#include <rimage/__/ext_manifest.h>`

- Fail2.2 SOF FW misses `#include <rimage/__/ext_manifest.h>`

Fail2.1 and Fail2.2 are both addressed by brand new https://github.com/thesofproject/sof/pull/4682

Fail1 and Fail2.1 were new and caused by the recent addition of sof-logger and smex in https://github.com/thesofproject/sof/commit/f6c71c21d03b8111

Failure 2.2 was NOT new! It used to be "fixed" by a Zephyr-specific copy of rimage/__/ext_manifest.h in commit https://github.com/nashif/sof/blob/973de63bef8658ed/zephyr/include/rimage/sof/kernel/ext_manifest.h / https://github.com/zephyrproject-rtos/sof/pull/7 but this PR drops 973de63bef8!

This should be the very first time a "clean", unmodified SOF commit builds in Zephyr CI! Crossing fingers...

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**
